docs/asb/README.md

```diff
@@ -0,0 +1,129 @@
+
+Currently, there is no automated discovery for service providers running an ASB.
+A service provider has to manually provide the connection details to users that will run the CLI.
+
+[Libp2p addressing](https://docs.libp2p.io/concepts/addressing/) is used to identify a service provider by multi-address and peer-id.
+The Peer-ID is printed upon startup of the ASB.
+The multi-address typically consists of IP-address or URL (if DNS entry configured) of the service provider.
+
+Each service provider running an ASB can decide how/where to share these connection details.
+
+![Service Provider scenarios](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/comit-network/xmr-btc-swap/d2cf45d8b9f0c2e180cd85aa034f370965adc11c/docs/asb/diagrams/cli-asb-overview.puml)
+
+Eventually, more elaborate discovery mechanisms can be added.
+
+The **CLI** user can specify a service providers's multiaddress and peer-id with `--seller-addr` and `--seller-peer`, see `./swap --help` for details.
+
+### Setup Details
+
+In order to understand the different components of the ASB and CLI better here is a component diagram showcasing the ASB and CLI setup using public Bitcoin and Monero infrastructure:
+
+![Service Provider scenarios](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/comit-network/xmr-btc-swap/363ce1cdf6fe6478736ff91e1458d650c2319248/docs/asb/diagrams/cli-asb-components-asb-pub-nodes.puml)
+
+Contrary, here is a diagram that showcases an ASB running it's own blockchain infrastructure:
+
+![Service Provider scenarios](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/comit-network/xmr-btc-swap/363ce1cdf6fe6478736ff91e1458d650c2319248/docs/asb/diagrams/cli-asb-components-asb-self-hosted.puml)
+
+The diagram shows that the `asb` group (representing the `asb` binary) consists of three components:
+
+1. Monero Wallet
+2. Bitcoin Wallet
+3. ASB
+
+The `ASB` depicted in the diagram actually consists of multiple components (protocol impl, network communication, ...) that sums up the functionality to execute concurrent swaps in the role of Alice.
+
+#### Monero Wallet Setup
+
+The ASB uses the running Monero wallet RPC to create / open Monero wallets.
+Currently you cannot connect to an existing Monero wallet, but the ASB will create the wallet `asb-wallet` upon intial startup.
+In order to accept trades with a CLI you will have to send XMR to that wallet.
+The wallet's address is printed upon startup of the ASB.
+Currently the `asb-wallet` does not have a password.
+
+Upon startup of the ASB the `asb-wallet` is opened in the wallet RPC.
+You can then interact with the wallet RPC for basic wallet management as well.
+
+#### Bitcoin Wallet Setup
+
+The ASB has an internally managed Bitcoin wallet.
+The Bitcoin wallet in created upon initial startup and stored in the data folder of the ASB (configured through initial startup wizard).
+
+#### Price Discovery
```

Comment:
Not sure if "Discovery" is such a good word here. Maybe "Market making" would be a better headline?

Reply:
Yeah, I guess price discovery is a bit too high level here ( https://www.investopedia.com/terms/p/pricediscovery.asp ).

Happy to rename it to Market Making.